framework/auth/views.py

```diff
@@ -230,9 +239,16 @@ def send_confirm_email(user, email):
     except NoResultsFound:
         merge_target = None
 
+    campaign = campaigns.campaign_for_user(user)
+    # Choose the appropriate email template to use
+    if merge_target:
+        mail_template = mails.CONFIRM_MERGE
+    else:
```

Comment:
This should be `elif campaign:` (don't take for granted that `campaign` is truthy is `merge_target` isn't)